gwcs/tags/wcs.py

```diff
@@ -305,3 +220,51 @@ def assert_equal(cls, old, new):
         assert old.name == new.name
         for old_frame, new_frame in zip(old.frames, new.frames):
             helpers.assert_tree_match(old_frame, new_frame)
+
+
+class TemporalFrameType(GWCSType):
+    name = "temporal_frame"
+    requires = _REQUIRES
+    types = [TemporalFrame]
+    version = '1.0.0'
+
+    @classmethod
+    def to_tree(cls, frame, ctx):
+        import astropy.time
+
+        node = {}
+
+        node['name'] = frame.name
+
+        if frame.axes_order != (0, 1):
+            node['axes_order'] = list(frame.axes_order)
+
+        if frame.axes_names is not None:
+            node['axes_names'] = list(frame.axes_names)
+
+        if frame.reference_frame is not None:
+            if frame.reference_frame is not astropy.time.Time:
+                raise ValueError("Can not save reference_frame unless it's Time")
+
+        if frame.reference_position is not None:
+            node['reference_time'] = yamlutil.custom_tree_to_tagged_tree(
+                frame.reference_position, ctx)
+
+        if frame.unit is not None:
+            node['unit'] = yamlutil.custom_tree_to_tagged_tree(
+                list(frame.unit), ctx)
```

Comment:
Shouldn't  `reference_time` be serialized?

Reply:
it is, the constructor for `TemporalFrame` takes `reference_time` as a kwarg but passes it as `reference_position` to the base class. So it is serialized above.